test/OpenTelemetry.Tests/Logs/BatchLogRecordExportProcessorTests.cs

```diff
@@ -43,14 +43,14 @@ public void StateValuesAndScopeBufferingTest()
         Assert.NotNull(logRecord.AttributeStorage);
         Assert.NotNull(logRecord.ILoggerData.BufferedScopes);
 
-        KeyValuePair<string, object> actualState = logRecord.StateValues[0];
+        KeyValuePair<string, object?> actualState = logRecord.StateValues[0];
 
         Assert.Same("Value", actualState.Key);
         Assert.Same("Hello world", actualState.Value);
 
         bool foundScope = false;
 
-        logRecord.ForEachScope<object>(
+        logRecord.ForEachScope<object?>(
             (s, o) =>
             {
                 foundScope = ReferenceEquals(s.Scope, exportedItems);
```

Comment:
Unrelated to your changes, but this looks like a test bug.

The foreach method is setting a boolean. This means only the last scope inspected is recorded.
I think this instead should be using an on-equals `|=`

```suggestion
                foundScope |= ReferenceEquals(s.Scope, exportedItems);
```